tests/integration/test_charm.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+# Copyright 2024 Kimonas Sotirchos
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    # Build and deploy charm from local source folder
+    charm = await ops_test.build_charm(".")
+
+    if ops_test.model is None:
+        logger.error("ops_test.model is not initialized!")
+        assert False
+
+    # Deploy the charm and wait for active/idle status
+    await asyncio.gather(
+        ops_test.model.deploy(charm, application_name=APP_NAME),
```

Comment:
nit: It's fine using `asyncio.gather()` although we usually just do `await ..`

Reply:
Updated to explicitly `await` each of the commands
ccabeab

Reply:
198aa9d